src/System.Net.Sockets/tests/FunctionalTests/NetworkStreamTest.netcoreapp.cs

```diff
@@ -395,6 +395,37 @@ public async Task ReadAsync_CancelPendingRead_DoesntImpactSubsequentReads()
             });
         }
 
+        [Fact]
+        public async Task WriteAsync_CancelPendingWrite_SucceedsOrThrowsOperationCanceled()
+        {
+            await RunWithConnectedNetworkStreamsAsync(async (server, client) =>
+            {
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(() => client.WriteAsync(new byte[1], 0, 1, new CancellationToken(true)));
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => { await client.WriteAsync(new Memory<byte>(new byte[1]), new CancellationToken(true)); });
+
+                byte[] hugeBuffer = new byte[100_000_000];
+                Exception e;
+
+                var cts = new CancellationTokenSource();
+                Task t = client.WriteAsync(hugeBuffer, 0, hugeBuffer.Length, cts.Token);
+                cts.Cancel();
+                e = await Record.ExceptionAsync(async () => await t);
```

Comment:
What is the `Record` type?? Where is it defined.

Reply:
It's part of xunit's asserts library.  I'm not sure why it was designed like this rather than as `Assert.RecordExceptionAsync` or something like that, but that's essentially what it is.  It just catches and returns an exception that was thrown.